omegaconf/base.py

```diff
@@ -542,10 +552,20 @@ def _wrap_interpolation_result(
         wrapped._set_flag("readonly", True)
         return wrapped
 
-    def _resolve_node_interpolation(
-        self,
-        inter_key: str,
-    ) -> "Node":
+    def _validate_not_dereferencing_to_parent(self, node: Node, target: Node) -> None:
+        parent: Optional[Node] = node
+        while parent is not None:
+            if parent is target:
+                self._format_and_raise(
+                    key=node._key(),
+                    value=None,
+                    cause=InterpolationResolutionError(
+                        "Interpolation to parent node detected"
+                    ),
+                )
+            parent = parent._get_parent()
```

Comment:
There's a potential for an infinite loop here (in case of accidental cycles in the parent graph) => can you add a safety check? (yay, another memo!)

Reply:
Hmm yeah it may be tricky to trigger it without something else blowing up. But since you wrote the code why not keep it?
(you can always write a test that directly calls `_validate_not_dereferencing_to_parent()`)

Main reason why I'm pointing it out is that it's an infinite loop and not a recursion error (and stuck programs suck).

Reply:
Agreed, 10 mins is plenty :)